rancher/rancher
# [2.7] Provide Internal & External IP addresses to node agent

## Issue: https://github.com/rancher/rancher/issues/36025
 
## Problem
When deploying clusters which have `generate_serving_certificates` enabled for the kubelet service, certificate files will be deployed to the downstream node. Depending on the infrastructure provider, these certificates may be named using the public or private IP address of the node. For example, a DO VM will have its certificate files named using the external IP address, while a VM on AWS will have its certificate files named using the private IP address. This is a problem for the node agent, which now needs to interrogate these certificate files to determine if they need to be refreshed. 
 
## Solution
Provide both the public and private IP addresses to the node agent, then check both possible file locations when performing cert validity checks. 
 
## Testing
+ Create a cluster with `generate_serving_certificate` enabled for the kubelet service. 
+ wait for the node to become available 
+ Find the pod running the node agent, and run the following command: `docker logs <NODE_AGENT_CONTAINER> | head`
+ Look in the logs for a line which starts with `INFO: Environment:`, and fine the `CATTLE_ADDRESS` and `CATTLE_INTERNAL_ADDRESS` values. Ensure they equal the actual external and internal IP address of the node

## Engineering Testing
### Manual Testing
I've done the above

I've also run through the existing test steps for the parent issue / PR, and did not observe any refreshes during restarts of rancher or during set periods of time (i.e. as new plans were requested) 

### Automated Testing
n/a

## QA Testing Considerations
Some providers do not set internal IP addresses, in which case the `CATTLE_INTERNAL_ADDRESS` value will be empty
 
### Regressions Considerations
This will not introduce any regressions as this logic is only used when refreshing the kubelet serving certificates
